File: tiq-e/strings/6/main.go
package main

import (
	"fmt"
	"math"
	"time"
)

func myAtoi(str string) int {
	sign, ret, i := 1, 0, 0

	for ; i < len(str) && str[i] == ' '; i++ {
	}

	if i < len(str) && (str[i] == '+' || str[i] == '-') {
		if str[i] == '-' {
			sign = -1
		}
		i++
	}
	for ; i < len(str) && '0' <= str[i] && str[i] <= '9'; i++ {
		ret = ret*10 + int(str[i]-'0')
		if sign*ret < math.MinInt32 {
			return math.MinInt32
		}
		if sign*ret > math.MaxInt32 {
			return math.MaxInt32
		}
	}
	return sign * ret
}

func main() {
	s := "42"

	var start = time.Now()
	var result = myAtoi(s)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}

File: tiq-e/strings/6/main.ts
import {performance} from 'perf_hooks';


function myAtoi(str: string): number {
    let filter = '0123456789+- '
    let res = 0
    let sign = 1
    for (let char of str) {
        let index = filter.indexOf(char)
        if (index != -1) {
            if (char == " ") continue
            if (filter[10] == "+") filter = filter.slice(0, 10)
            if (char == "+") continue
            if (char == "-") {
                sign = -sign;
                continue
            }
            res = res * 10 + index
        } else {
            break;
        }
    }
    res = res * sign
    if (res > 2 ** 31 - 1) res = 2 ** 31 - 1
    else if (res < -(2 ** 31)) res = -(2 ** 31)
    return res
}


const s = "42";

const start = performance.now();
const result = myAtoi(s);
const end = performance.now();
console.log('Duration:', end - start);
console.log('Result:', result)
